.github/workflows/ci_helm.yaml

```
@@ -0,0 +1,65 @@
+name: Build and Push Helm chart
+on:
+  workflow_call:
+    inputs:
+      project_name:
+        required: true
+        type: string
+      aws_account_id:
+        required: true
+        type: string
+      aws_region:
+        required: true
+        type: string
+
+jobs:
+  build_and_push:
+    name: Build and Push
+    runs-on: ["self-hosted", "can-run-small-job"]
+    timeout-minutes: 1
+    outputs:
+      role_arn: ${{ steps.set_roles.outputs.role_arn }}
+      role_session_name: ${{ steps.set_roles.outputs.role_session_name }}
+      chart_tag: ${{ steps.form_chart_tag.outputs.value }}
+    steps:
+      - id: set_roles
+        run: |
+          ROLE_ARN="arn:aws:iam::${{ inputs.aws_account_id }}:role/github_actions_ecr_${{ inputs.project_name }}"
+          ROLE_SESSION_NAME="github_actions_ci_${{ inputs.project_name }}"
+          echo "::set-output name=role_arn::${ROLE_ARN}"
+          echo "::set-output name=role_session_name::${ROLE_SESSION_NAME}"
+      - name: Extract branch name
+        run: echo "::set-output name=branch::${GITHUB_REF#refs/heads/}"
+        id: extract_branch
+      - name: Form Chart Tag name
+        id: form_chart_tag
+        uses: haya14busa/action-cond@v1
+        with:
+          cond: ${{ github.event_name == 'pull_request' }}
+          if_true: pr-${{ github.event.pull_request.number }}-${{ github.run_number }}
+          if_false: ${{ steps.extract_branch.outputs.branch }}-${{ github.run_number }}
```

Эту шляпу я давно придумал и она должна умереть :)
```suggestion
      - name: Form Chart Tag name
        id: form_chart_tag
        run: |
          TAG=${GITHUB_REF#refs/heads/}-${{ github.run_number }}-${{ github.run_attempt }}
          if [[ ${{ github.event_name }} == "pull_request" ]] then;
              TAG=pr-${{ github.event.pull_request.number }}-${{ github.run_number }}-${{ github.run_attempt }}
          fi
          echo "::set-output..."
```